### DISCUSSION TITLE
NBD Block device

### DISCUSSION THREAD

**Discussion by vkornilyev1988** [ipxe/ipxe#1157] [General]
Tell me how I can write a protocol for sanboot nbd so that, like AOE/ISCSI, I can load it. by nbd? Exmaple sanboot nbd:1.1.1.1:10890?

**Comment by NiKiZe** [comment]
Your wording sounds like a demand, you will have to do at least basic research yourself and can then ask specific questions as discussions (not issues!)

However you might be interested in #393 (another good thing to do is using the search function)

**Comment by vkornilyev1988** [comment]
Hi, Yes cool ! I’m writing a diskless system where even loading is faster on the nbd protocol. I used a driver for Windows 10/11 and everything worked great, but the bootloader was missing and now I see that I can try these fixes that you indicated above. Everyone is asking for money for a diskless system, but I want to make it free and put it on GitHub. It was NBD that allowed me to use everything I needed and comply with the Ethical Issues of Free Software. Thank you! But it would be cool if this were added nbd protocol to the ipxe branch . I also implemented the wnbd table on an equal basis with ibft and abft. Thank you!

**Reply by NiKiZe** [reply]
How is the handover between bootloader and OS done? There is a standard for this iSCSI and iBFT going down the NBD path will give several pain points down the line, there is also security implications which iPXE don't really want to deal with.

NFS is supported as well, but forced disabled for EFI builds with SB enabled due to issues found during review.

**Comment by vkornilyev1988** [comment]
I want to ask one more thing. Is work being done on the secure boot by windows in uefi?

**Reply by NiKiZe** [reply]
Sorry, I will be blunt here, Was there to many search results to look thru?

**Comment by vkornilyev1988** [comment]
Thank's ! Yes, there is a demand for computer clubs where they use CCBOOT. Due to the fact that the iscsi are only available for Windows server. Computer clubs pay rent for each computer in order to be able to download each computer. Please note that I have implemented wbft (Windows NBD firmware table) using ipxe table for nbd. This table replaces ibft, but only for the nbd protocol. Accordingly, work with the Windows registry and drivers has already been implemented. You are very helpful. But I think I will have a problem with secure boot in the future(

**Reply by NiKiZe** [reply]
iSCSI initiator is available in most versions of Windows. There is several free and open source variants of iSCSI serverrs.

**Comment by vkornilyev1988** [comment]
Accordingly, my goal is to give people more freedom to choose systems such as CCBOOT. I really want this to be available to everyone and in the form of source codes like GPLv3.))))

**Comment by vkornilyev1988** [comment]
for free)
